lib/kino.ex

```diff
@@ -193,4 +212,60 @@ defmodule Kino do
   def nothing() do
     :"do not show this result in output"
   end
+
+  @doc """
+  Starts a process under the Kino supervisor.
+
+  The process is automatically terminated when the current process
+  terminates or the current cell reevaluates.
+  """
+  @spec start_child(
+          Supervisor.child_spec()
+          | {module(), term()}
+          | module()
+        ) :: DynamicSupervisor.on_start_child()
+  def start_child(child_spec) do
+    # Starting a process that calls Kino.start_child/1 in its init
+    # would block forever, so we don't allow nesting
+    if Kino.DynamicSupervisor in Process.get(:"$ancestors", []) do
+      raise "could not start #{inspect(child_spec)} using Kino.start_child/1," <>
+              " because the current process has been started with Kino.start_child/1." <>
+              " Please move the nested start outside and pass the result as an argument to this process"
+    end
+
+    %{start: start} = child_spec = Supervisor.child_spec(child_spec, [])
+    parent = self()
+    child_spec = %{child_spec | start: {Kino, :__start_override__, [start, parent]}}
+    DynamicSupervisor.start_child(Kino.DynamicSupervisor, child_spec)
+  end
+
+  @doc false
+  def __start_override__({mod, fun, args}, parent) do
+    # We switch the group leader, so that the newly started
+    # process gets the same group leder as the caller
+    initial_gl = Process.group_leader()
+    gl = Process.info(parent)[:group_leader]
+
+    Process.group_leader(self(), gl)
+
+    try do
+      apply(mod, fun, args)
+      |> case do
+        {:ok, pid} = resp -> {:ok, resp, pid}
+        {:ok, pid, _info} = resp -> {:ok, resp, pid}
+        resp -> {:error, resp}
+      end
```

Comment:
Nitpick, return this:

```suggestion
      |> case do
        {:ok, pid} = resp -> {resp, pid}
        {:ok, pid, _info} = resp -> {resp, pid}
        resp -> {resp, nil}
      end
```

Then below you do:

```elixir
    {resp, pid} ->
      pid && Kino.Bridge.object_monitor(pid, parent, Kino.Terminator.cross_node_name(), {:terminate, pid})
      resp
```